modules/statistics/CallStats.js

```diff
@@ -235,330 +403,300 @@ CallStats.prototype.associateStreamWithVideoTag = function(
- *
- * @param overallFeedback an integer between 1 and 5 indicating the
- * user feedback
- * @param detailedFeedback detailed feedback from the user. Not yet used
- */
-CallStats.prototype.sendFeedback = tryCatch(
-function(overallFeedback, detailedFeedback) {
-    if (!CallStats.feedbackEnabled) {
-        return;
+    /* eslint-disable no-invalid-this */
+    /**
+     * Sends the given feedback through CallStats.
+     *
+     * @param {string} conferenceID the conference ID for which the feedback
+     * will be reported.
+     * @param overallFeedback an integer between 1 and 5 indicating the
+     * user feedback
+     * @param detailedFeedback detailed feedback from the user. Not yet used
+     */
+    static sendFeedback(conferenceID, overallFeedback, detailedFeedback) {
+        if (backend) {
+            backend.sendUserFeedback(
+                conferenceID, {
+                    userID: CallStats.userID,
+                    overall: overallFeedback,
+                    comment: detailedFeedback
+                });
+        } else {
+            logger.error('Failed to submit feedback to CallStats - no backend');
+        }
     }
 
-    callStats.sendUserFeedback(this.confID, {
-        userID: this.userID,
-        overall: overallFeedback,
-        comment: detailedFeedback
-    });
-});
+    /* eslint-enable no-invalid-this */
+
+    /* eslint-disable max-params */
+    /**
+     * Reports an error to callstats.
+     *
+     * @param {CallStats} [cs]
+     * @param type the type of the error, which will be one of the wrtcFuncNames
+     * @param e the error
+     * @param pc the peerconnection
+     * @private
+     */
+    static _reportError(cs, type, e, pc) {
+        let error = e;
+
+        if (!error) {
+            logger.warn('No error is passed!');
+            error = new Error('Unknown error');
+        }
+        if (CallStats.initialized) {
+            backend.reportError(pc, cs && cs.confID, type, error);
+        } else {
+            CallStats.reportsQueue.push({
+                type: reportType.ERROR,
+                data: {
+                    error,
+                    pc,
+                    type
+                }
+            });
+        }
 
-/* eslint-enable no-invalid-this */
+        // else just ignore it
+    }
 
-/**
- * Reports an error to callstats.
- *
- * @param type the type of the error, which will be one of the wrtcFuncNames
- * @param e the error
- * @param pc the peerconnection
- * @private
- */
-CallStats._reportError = function(type, e, pc) {
-    let error = e;
+    /* eslint-enable max-params */
 
-    if (!error) {
-        logger.warn('No error is passed!');
-        error = new Error('Unknown error');
+    /**
+     * Notifies CallStats that getUserMedia failed.
+     *
+     * @param {Error} e error to send
+     * @param {CallStats} cs callstats instance related to the error (optional)
+     */
+    static sendGetUserMediaFailed(e, cs) {
```

Comment:
I think all static methods should be together, sorted amongst themselves, then instance methods, also sorted.